client/mm/libxc/mexc.go

```diff
@@ -0,0 +1,3248 @@
+		return fmt.Errorf("failed to get exchange info: %w", err)
+	}
+
+	// Populate symbols cache
+	syms := make(map[string]struct{}, len(info.Symbols))
+	symbolMeta := make(map[string]*mxctypes.Symbol, len(info.Symbols))
+
+	for _, s := range info.Symbols {
+		symbol := strings.ToUpper(s.Symbol)
+		syms[symbol] = struct{}{}
+		symbolMeta[symbol] = &s
+	}
+
+	m.symbolsMtx.Lock()
+	m.symbols = syms
+	m.symbolsMtx.Unlock()
+
+	m.symbolMetaMtx.Lock()
+	m.symbolMeta = symbolMeta
+	m.symbolMetaMtx.Unlock()
+
+	// Fetch API-tradeable symbols immediately after exchange info
+	// This ensures Markets() will have the tradeable list when called
+	if err := m.getDefaultSymbols(ctx); err != nil {
+		m.log.Errorf("Failed to fetch API-tradeable symbols: %v", err)
+		// Don't fail - Markets() will use optimistic default (assume all tradeable)
+	}
+
+	return nil
+}
+
+func (m *mexc) symbolExists(sym string) bool {
+	m.symbolsMtx.RLock()
+	_, ok := m.symbols[strings.ToUpper(sym)]
+	m.symbolsMtx.RUnlock()
+	return ok
+}
+
+// getSymbolMeta returns the symbol metadata for a given symbol.
+func (m *mexc) getSymbolMeta(symbol string) *mxctypes.Symbol {
+	m.symbolMetaMtx.RLock()
+	defer m.symbolMetaMtx.RUnlock()
+	return m.symbolMeta[strings.ToUpper(symbol)]
+}
+
+// getFilterValue returns the value of a specific filter type for a symbol.
+func (m *mexc) getFilterValue(symbol, filterType string) string {
+	meta := m.getSymbolMeta(symbol)
+	if meta == nil {
+		return ""
+	}
+	for _, filter := range meta.Filters {
+		if filter.FilterType == filterType {
+			switch filterType {
+			case "PRICE_FILTER":
+				return filter.TickSize
+			case "LOT_SIZE":
+				return filter.StepSize
+			case "MIN_NOTIONAL":
+				return filter.MinNotional
+			case "MIN_QTY":
+				return filter.MinQty
+			case "MAX_QTY":
+				return filter.MaxQty
+			}
+		}
+	}
+	return ""
+}
+
+// snapPrice snaps a price to the tick size for a symbol.
+func (m *mexc) snapPrice(symbol string, price float64) float64 {
+	tickSizeStr := m.getFilterValue(symbol, "PRICE_FILTER")
+	if tickSizeStr == "" {
+		return price
+	}
+	tickSize, err := strconv.ParseFloat(tickSizeStr, 64)
+	if err != nil {
+		return price
+	}
+	return float64(int64(price/tickSize+0.5)) * tickSize
+}
+
+// snapQuantity snaps a quantity to the step size for a symbol.
+func (m *mexc) snapQuantity(symbol string, qty float64) float64 {
+	stepSizeStr := m.getFilterValue(symbol, "LOT_SIZE")
+	if stepSizeStr == "" {
+		return qty
+	}
+	stepSize, err := strconv.ParseFloat(stepSizeStr, 64)
+	if err != nil {
+		return qty
+	}
+	return float64(int64(qty/stepSize)) * stepSize
+}
+
+// validateMinNotional checks if the order meets the minimum notional value.
+func (m *mexc) validateMinNotional(symbol string, price, qty float64) bool {
+	minNotionalStr := m.getFilterValue(symbol, "MIN_NOTIONAL")
+	if minNotionalStr == "" {
+		return true
+	}
+	minNotional, err := strconv.ParseFloat(minNotionalStr, 64)
+	if err != nil {
+		return true
+	}
```

Comment:
Is this sometimes not there? Can you log the error if ParseFloat fails? Seems like that would be false as well?